src/query/list_file.c

```diff
@@ -392,6 +394,55 @@ qfile_list_cache_cleanup (THREAD_ENTRY * thread_p)
 }
 #endif
 
+int
+qcache_get_new_ht_no (THREAD_ENTRY * thread_p)
+{
+  int ht_no = -1;
+
+  if (QFILE_IS_LIST_CACHE_DISABLED)
+    {
+      return NO_ERROR;
+    }
+
+  if (csect_enter (thread_p, CSECT_QPROC_LIST_CACHE, INF_WAIT) != NO_ERROR)
+    {
+      return ER_FAILED;
+    }
+
+  if (qfile_List_cache.next >= 0)
+    {
+      ht_no = qfile_List_cache.next;
+      qfile_List_cache.next = qfile_List_cache.free_list[qfile_List_cache.next];
+    }
+
+  csect_exit (thread_p, CSECT_QPROC_LIST_CACHE);
+
+  return ht_no;
+}
+
+void
+qcache_free_ht_no (THREAD_ENTRY * thread_p, int ht_no)
+{
+  if (QFILE_IS_LIST_CACHE_DISABLED)
+    {
+      return;
+    }
+
+  if (csect_enter (thread_p, CSECT_QPROC_LIST_CACHE, INF_WAIT) != NO_ERROR)
+    {
+      return;
+    }
+
+  if (ht_no >= 0)
```

Comment:
It looks good to write 'if (ht_no >= 0)' before csect_enter() .

Reply:
Ok, I will fix it.